### DISCUSSION TITLE
should i rewrite the editor??

### DISCUSSION THREAD

**Discussion by raynepaws** [cytrinc/bark-coding#109] [Admin General]
from reading the title, you probably think i'm a lunatic, but hear me out.

currently, it's very complicated to add new blocks and get them to function properly, as shown through the turmoil we've had trying to get the "when green button clicked" to run code. so, here's my idea:

i will completely rewrite the editor, totally abandoning blockly. i'll make it as simple as possible to function and add new blocks. it'll be made specially for bark, so we don't have to play along to blockly's rules

if you want, i can create a basic idea with a few blocks to show to you guys. i've got nothing to do this weekend so it would be a good project to spend my time on. i feel this would be very beneficial for us all and make life a lot easier

**Comment by mariocraft987** [comment]
you could do that in a personal fork then create a pull request while we inspect it

**Reply by raynepaws** [reply]
alr, i'll start asap

**Comment by nmsderp** [comment]
in scratch you have to go through a very similar process, and tbh you can add blocks in like 5 minutes, blockly is also very optimized

**Reply by raynepaws** [reply]
i would argue it's not. there's so much stuff that people like me just don't understand. and, the documentation will be adequate so it's easy to learn and takes less time to add blocks. trust me

**Comment by rubiidev18alt** [comment]
you can make a new repo to showcase us your progress on this new editor!

edit: nvm, you already did that, but enable discussions on your repo so we can see your devlog! + invite us!

**Reply by mariocraft987** [reply]
Thats exactly what I said
> you could do that in a personal fork then create a pull request while we inspect it

**Reply by raynepaws** [reply]
alr! i didn't think of a devlog, good idea!

**Comment by raynepaws** [comment]
sorry i never got to this

**Reply by mariocraft987** [reply]
k
